src/gauging.jl

```diff
@@ -206,25 +155,19 @@ function vidal_itn_isometries(
   return isometries
 end
 
-"""Function to measure the 'canonicalness' of a state in the Vidal Gauge"""
-function vidal_itn_canonicalness(ψ::ITensorNetwork, bond_tensors::DataGraph)
+"""Function to measure the 'distance' of a state from the Vidal Gauge"""
+function gauge_error(ψ::ITensorNetwork, bond_tensors::DataGraph)
   f = 0
-
   isometries = vidal_itn_isometries(ψ, bond_tensors)
-
   for e in keys(isometries)
-    LHS = isometries[e] / sum(diag(isometries[e]))
-    id = dense(delta(inds(LHS)))
-    id /= sum(diag(id))
-    f += 0.5 * norm(id - LHS)
+    lhs = isometries[e]
+    f += message_diff(ITensor[lhs], ITensor[dense(delta(inds(lhs)))])
   end
 
   return f / (length(keys(isometries)))
 end
 
-"""Function to measure the 'canonicalness' of a state in the Symmetric Gauge"""
-function symmetric_itn_canonicalness(ψ::ITensorNetwork, pψψ::PartitionedGraph, mts)
-  ψ_vidal, bond_tensors = symmetric_to_vidal_gauge(ψ, pψψ, mts)
-
-  return vidal_itn_canonicalness(ψ_vidal, bond_tensors)
+function gauge_error(ψ::ITensorNetwork, bp_cache::BeliefPropagationCache)
```

Comment:
I think we should pass the cache as a keyword argument `cache!`.

Reply:
But in this case the function is just returning the error associated with `psi` and its bp_cache. So having the ability to update the cache or have it build a new cache doesn't really make sense to me here.

It probably shouldn't be called `gauge_error` at all and is really more of a bp function for how good a `bp_cache` is. I have now removed the function.